### PR TITLE
fix(certs): respect the secret name provided in the executor

### LINKED ISSUE #234: add e2e for default CA bundle work added for supporting outbound http apps

### PR DESCRIPTION
**Pull request by adamreese**
- fixes a bug where using custom name doesn't create the secret
- add test for ensuring ca certificate secret

Closes #234

### CODE REVIEW

File: e2e/default_test.go [hunk 1]
```diff
@@ -103,22 +146,21 @@ func newSpinAppCR(name, image string) *spinapps_v1alpha1.SpinApp {
 			Executor: "containerd-shim-spin",
 		},
 	}
-	return testSpinApp
 }
 
-func newContainerdShimExecutor(namespace string) *spinapps_v1alpha1.SpinAppExecutor {
```
Comment (endocrimes) [inline e2e/default_test.go:109]: Plz revert (helpers should at least take the namespace for easier conversion to per-test namespaces later)

**Comment by endocrimes** [comment]
We probably need to also add a test that we won't overwrite an existing secret - I thought we previously made the assumption that if you provided a name you wanted to use your own certs (so this becomes somewhat racy and risks breaking that)

File: e2e/default_test.go (hunk 1 above)
Comment (adamreese) [reply e2e/default_test.go:109]: Got it. I was making it consistent with the other helper

**Comment by adamreese** [comment]
>We probably need to also add a test that we won't overwrite an existing secret

I agree. I didn't attempt that with this PR because it would change a lot of the current pattern. I think it would be better in a new PR?

File: e2e/default_test.go (hunk 1 above)
Comment (endocrimes) [reply e2e/default_test.go:109]: Oh cool (we should change the other one when we next touch it 😅 )

**Comment by adamreese** [comment]
Any other blockers for getting this merged?

**Review by calebschoepp — APPROVED** [review]
No blockers here

**Pull request merged**